File: foundation/ace/ace_engine_lite/frameworks/examples/showcase/src/main/js/default/pages/component/qrcode/event/event.js
/*
 * Copyright (c) 2020 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { goPage, routePage } from "../../../../common/js/general";
export default {
  data: {
    qr_color: ""
  },
  ...goPage("pages/component/qrcode/index/index"),
  clickEvent() {
    this.qr_color = "#fa8072";
  },
  longpressEvent() {
    this.qr_color = "#87ceeb";
  },
  swipeEvent(e) {
    this.qr_color = '#50bf1f';
  }
};

File: foundation/ace/ace_engine/frameworks/core/components/declaration/common/event.cpp
/*
 * Copyright (c) 2021 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include "core/components/declaration/common/event.h"

#include "core/event/ace_event_helper.h"
#include "core/event/back_end_event_manager.h"

namespace OHOS::Ace {

void EventParam::FireEvent(const WeakPtr<PipelineContext>& context)
{
    auto eventContext = context.Upgrade();
    if (!eventContext) {
        return;
    }
    auto event = AceAsyncEvent<void()>::Create(eventMarker, context);
    event();
}

void TouchEventParam::FireEvent(const WeakPtr<PipelineContext>& context)
{
    auto eventContext = context.Upgrade();
    if (!eventContext) {
        return;
    }
    auto event = AceAsyncEvent<void(const TouchEventInfo&)>::Create(eventMarker, context);
    event(touchEvent);
}

} // namespace OHOS::Ace